app/controllers/users.js

```diff
@@ -1,26 +1,58 @@
 const util = require('util');
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
 
-const User = require('../models').users;
 const logger = require('../logger');
 const error = require('../errors');
-const { hashPassword } = require('../services/users');
+const { hashPassword, findByEmail, createUser } = require('../services/users');
 
-const { userCreationError } = error;
+const { userCreationError, userLoginError } = error;
 
-exports.postUser = (req, res, next) => {
+exports.signUpUser = (req, res, next) => {
   const { first_name: firstName, last_name: lastName, email, password } = req.body;
-  return hashPassword(password)
-    .then(() =>
-      User.create({ firstName, lastName, email, password })
-        .then(user => {
-          logger.info(`Usuario creado para: ${firstName} ${lastName}`);
-          return user;
+  return findByEmail(email)
+    .then(foundUser => {
+      if (foundUser) {
+        throw userCreationError(`User for ${email} already exists!`);
+      } else {
+        return hashPassword(password)
+          .then(hashedPassword =>
+            createUser(firstName, lastName, email, hashedPassword)
+              .then(user => {
+                logger.info(`New user created for: ${firstName} ${lastName}`);
+                res.status(201).send(user);
+              })
+              .catch(err => {
+                logger.error(util.inspect(err));
+                throw userCreationError(err.message);
+              })
+          )
+          .then(response => res.status(201).send(response))
```

Comment:
Mmm this it's wrong. Delete this line

Reply:
You're right, it wasn't necesary. Thanks